kouncil-backend/src/main/java/com/consdata/kouncil/track/TrackController.java

```diff
@@ -108,9 +114,19 @@ public TrackController(KafkaConnectionService kafkaConnectionService) {
                                     .build());
                         }
                     }
+                    log.debug("TRACK90 poll completed topic={}, candidates.size={}", t, candidates.size());
+                    if (!candidates.isEmpty()) {
+                        if (Strings.isNotBlank(asyncHandle)) {
+                            candidates.sort(Comparator.comparing(TopicMessage::getTimestamp));
+                            String destination = "/topic/track/" + asyncHandle;
+                            log.debug("TRACK91 async batch send topic={}, destination={}, size={}", t, destination, candidates.size());
+                            template.convertAndSend(destination, candidates);
```

Comment:
Nie jestem pewien jak to działa, ale tu się nic nie wywali, jeśli klient w międzyczasie zamknął przeglądarkę, albo ten komponent? A jeśli wyszedł/zamknął, i tu się nam coś wywali w związku z tym, to może w ogóle warto by wyjście z tego while'a? Choć potencjalnie jeśli się wywali, to i tak samo wyjdzie :D no w każdym razie, zastanawiam się, czy tu nie potrzeba jeszcze jakiejś niewielkiej dodatkowej obsługi, jeśli coś poszło nie tak

Reply:
zbadam temat

Reply:
działa to w ten sposób, że jak klient zniknie, to nie lecą błądy, po prostu kończy się przetwarzanie i topic jest usuwany. Inna sprawa jest tak, że można monitorować i przerwać przetwarzanie w backendzie jak nie ma klienta, ale na to jest osobne zadanie.